coldfront/core/project/management/commands/deactivate_ica_projects.py

```diff
@@ -0,0 +1,236 @@
+                      f'ICA project."'
+
+            self.stdout.write(self.style.WARNING(message))
+
+        else:
+            # Set the value for the Project.
+            set_project_allocation_value(project, updated_su)
+
+            # Create a transaction to record the change.
+            ProjectTransaction.objects.create(
+                project=project,
+                date_time=current_date,
+                allocation=updated_su)
+
+            # Set the reason for the change in the newly-created historical object.
+            allocation_objects.allocation_attribute.refresh_from_db()
+            historical_allocation_attribute = \
+                allocation_objects.allocation_attribute.history.latest("id")
+            historical_allocation_attribute.history_change_reason = reason
+            historical_allocation_attribute.save()
+
+            # Do the same for each ProjectUser.
+            allocation_attribute_type = AllocationAttributeType.objects.get(
+                name="Service Units")
+            for project_user in project.projectuser_set.all():
+                user = project_user.user
+                # Attempt to set the value for the ProjectUser. The method returns whether
+                # it succeeded; it may not because not every ProjectUser has a
+                # corresponding AllocationUser (e.g., PIs). Only proceed with further steps
+                # if an update occurred.
+                allocation_updated = set_project_user_allocation_value(
+                    user, project, updated_su)
+                if allocation_updated:
+                    # Create a transaction to record the change.
+                    ProjectUserTransaction.objects.create(
+                        project_user=project_user,
+                        date_time=current_date,
+                        allocation=updated_su)
+                    # Set the reason for the change in the newly-created historical object.
+                    allocation_user = \
+                        allocation_objects.allocation.allocationuser_set.get(user=user)
+                    allocation_user_attribute = \
+                        allocation_user.allocationuserattribute_set.get(
+                            allocation_attribute_type=allocation_attribute_type,
+                            allocation=allocation_objects.allocation)
+                    historical_allocation_user_attribute = \
+                        allocation_user_attribute.history.latest("id")
+                    historical_allocation_user_attribute.history_change_reason = reason
+                    historical_allocation_user_attribute.save()
+
+            message = f'Successfully reset SUs for {project.name} ' \
+                      f'and its users, updating {project.name}\'s SUs from ' \
+                      f'{current_allocation} to {updated_su}. The reason ' \
+                      f'was: "{reason}".'
+
```

Comment:
I neglected to mention this, but we should also zero out the relevant `AllocationAttributeUsage` and `AllocationUserAttributeUsages`.

There should be some helper functions or code to reference for doing this.

Reply:
Do we want to update the historical objects when we do this as well?

Reply:
No, setting the reason in the attribute should be good enough.

(We're not setting these reasons elsewhere in the code, though it might be worth doing in the future.)